cli/src/pcluster/config/cluster_config.py

```diff
@@ -1629,17 +1629,127 @@ def __init__(
         )
-        self.__instance_type_info = None
         self.efa = efa or Efa(enabled=False, implied=True)
         self.schedulable_memory = Resource.init_param(schedulable_memory)
 
+    @property
+    @abstractmethod
+    def disable_simultaneous_multithreading_manually(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def disable_simultaneous_multithreading_via_cpu_options(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def pass_cpu_options_in_launch_template(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def max_network_interface_count(self) -> int:
+        pass
+
+    @property
+    @abstractmethod
+    def is_ebs_optimized(self) -> bool:
+        pass
+
+    @property
+    @abstractmethod
+    def instance_types(self) -> List[str]:
+        pass
+
+
+class FlexibleInstanceType(Resource):
+    """Represent an instance type listed under InstanceTypes under ComputeResources."""
+
+    def __init__(self, instance_type: str, **kwargs):
+        super().__init__(**kwargs)
+        self.instance_type = Resource.init_param(instance_type)
+
+
+class SlurmFlexibleComputeResource(_BaseSlurmComputeResource):
+    """Represents a Slurm Compute Resource with Multiple Instance Types."""
+
+    def __init__(self, instance_type_list: List[FlexibleInstanceType], **kwargs):
+        super().__init__(**kwargs)
+        self.instance_type_list = Resource.init_param(instance_type_list)
+
+    @staticmethod
+    def get_instance_type_info(instance_type) -> InstanceTypeInfo:
+        """Return instance type information."""
+        return AWSApi.instance().ec2.get_instance_type_info(instance_type)
+
+    @property
+    def instance_types(self) -> List[str]:
+        """Return list of instance type names in this compute resource."""
+        return [flexible_instance_type.instance_type for flexible_instance_type in self.instance_types]
+
+    @property
+    def disable_simultaneous_multithreading_manually(self) -> bool:
+        """Return true if simultaneous multithreading must be disabled with a cookbook script."""
+        return self.disable_simultaneous_multithreading
+
+    @property
+    def disable_simultaneous_multithreading_via_cpu_options(self) -> bool:
+        """CPU Options not supported for Compute Resources with Flexible Instance Types."""
+        return False
+
+    @property
+    def pass_cpu_options_in_launch_template(self) -> bool:
+        """CPU Options not supported for Compute Resources with Flexible Instance Types."""
+        return False
+
+    @property
+    def max_network_interface_count(self) -> int:
+        """Return max number of NICs for the compute resource.
+
+        In this case the Compute Resource may have multiple instance types, hence the instance-type with
+        the least MaxNetworkInterfaceCards value will be considered.
+        """
+        least_max_nics = SlurmFlexibleComputeResource.get_instance_type_info(
```

Comment:
if this method works for a instance type list with just 1 element, I will define it in the Base class so that it can be used for both SlurmFlexibleComputeResource and SlurmComputeResource